Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -220,17 +220,20 @@ private void InteractionSourceUpdated(InteractionSourceState state)
             {
                 if (controller.Interactions.ContainsKey(DeviceInputType.SpatialPointer) && controller.Interactions.GetDictionaryValueChanged(DeviceInputType.SpatialPointer))
                 {
-                    inputSystem?.Raise6DofInputChanged(controller.InputSource, controller.ControllerHandedness, controller.Interactions[DeviceInputType.SpatialPointer].InputAction, controller.Interactions[DeviceInputType.SpatialPointer].GetTransform());
+                    var data = controller.Interactions[DeviceInputType.SpatialPointer] as InteractionMapping<Tuple<Vector3, Quaternion>>;
+                    inputSystem?.Raise6DofInputChanged(controller.InputSource, controller.ControllerHandedness, controller.Interactions[DeviceInputType.SpatialPointer].InputAction, data.GetValue());
                 }
 
                 if (controller.Interactions.ContainsKey(DeviceInputType.SpatialGrip) && controller.Interactions.GetDictionaryValueChanged(DeviceInputType.SpatialGrip))
                 {
-                    inputSystem?.Raise6DofInputChanged(controller.InputSource, controller.ControllerHandedness, controller.Interactions[DeviceInputType.SpatialGrip].InputAction, controller.Interactions[DeviceInputType.SpatialGrip].GetTransform());
+                    var data = controller.Interactions[DeviceInputType.SpatialGrip] as InteractionMapping<Tuple<Vector3, Quaternion>>;
+                    inputSystem?.Raise6DofInputChanged(controller.InputSource, controller.ControllerHandedness, controller.Interactions[DeviceInputType.SpatialGrip].InputAction, data.GetValue());
                 }
 
                 if (controller.Interactions.ContainsKey(DeviceInputType.TouchpadTouch) && controller.Interactions.GetDictionaryValueChanged(DeviceInputType.TouchpadTouch))
                 {
-                    if (controller.Interactions[DeviceInputType.TouchpadTouch].GetBooleanValue())
+                    var data = controller.Interactions[DeviceInputType.TouchpadTouch] as InteractionMapping<bool>;
```

Comment:
Need TouchpadPress also

Reply:
Will check as I thought I had added that.